### PR TITLE
Fix flash sim write

### PR DESCRIPTION
**Pull request by de-nordic**
Fixes write for simulated flash devices with non-0xff erase value.

### CODE REVIEW

**Comment by de-nordic** [comment]
Problem identified by @Laczen . Thanks!

File: tests/drivers/flash_simulator/testcase.yaml [hunk 1]
```diff
@@ -2,3 +2,15 @@ tests:
...
     tags: driver
+  drivers.flash.flash_simulator.qemu_erase_value_0x00:
+    extra_args: DTC_OVERLAY_FILE=boards/qemu_x86_ev_0x00.overlay
+    platform_whitelist: qemu_x86
+    tags: driver
+  drivers.flash.flash_simulator.posix_erase_value_0x00:
+    extra_args: DTC_OVERLAY_FILE=boards/native_posix_ev_0x00.overlay
+    platform_whitelist: native_posix
+    tags: driver
+  drivers.flash.flash_simulator.posix_erase_value_0x00:
```
Comment (nvlsianpu) [inline tests/drivers/flash_simulator/testcase.yaml:13]:
```suggestion
  drivers.flash.flash_simulator.posix_64_erase_value_0x00:
```

**Review by nvlsianpu — CHANGES_REQUESTED** [review]
looks fine, just one nip on test name.

File: tests/drivers/flash_simulator/testcase.yaml (hunk 1 above)
Comment (de-nordic) [reply tests/drivers/flash_simulator/testcase.yaml]: Done.

**Review by nvlsianpu — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**